flink-runtime/src/main/java/org/apache/flink/runtime/state/filesystem/FsCheckpointMetadataOutputStream.java

```diff
@@ -162,4 +161,20 @@ public FsCompletedCheckpointStorageLocation closeAndFinalizeCheckpoint() throws
             }
         }
     }
+
+    static MetadataOutputStreamWrapper getOutputStreamWrapper(
+            final FileSystem fileSystem, final Path metadataFilePath) throws IOException {
+        try {
+            RecoverableWriter recoverableWriter = fileSystem.createRecoverableWriter();
+            if (fileSystem.exists(metadataFilePath)) {
+                throw new IOException("Target file " + metadataFilePath + " is already exists.");
```

Comment:
I think we should try to verify the file whether existed first. 
Moreover, the statement `"Target file " + metadataFilePath + " already exists."` looks better.

Reply:
👍  Got it.

Reply:
Updated. @Myasuka